### DISCUSSION TITLE
What is the status of support for Intel compilers?

### DISCUSSION THREAD

**Discussion by dkokron** [rizsotto/Bear#576] [General]
What is the status of support for Intel compilers (ifort, icc, ifx, icx)?

**Comment by rizsotto** [comment]
Not out of the box, but you can add any compiler to the config file to be recognized.

**Reply by dkokron** [reply]
The last time anyone tried to do what I'm looking for, they modified a file called main.py.in.  That file does not exist in bear-3.1.2.

Other attempts to add support for Intel compilers.
https://github.com/rizsotto/Bear/pull/138
https://github.com/rizsotto/Bear/pull/137
https://github.com/rizsotto/Bear/pull/253

**Reply by ewhac** [reply]
> Not out of the box, but you can add any compiler to the config file to be recognized.

Cool.  Could you please post a *working, concrete example* of such?  I have spent considerable time trying to work out the magic incantation to get some embedded development compilers recognized, without success.

**Reply by rizsotto** [reply]
just an update on this... there is new version where the compiler support had more attention and cleaned up the codebase.

The [compilers](https://github.com/rizsotto/Bear/tree/master/bear/src/semantic/interpreters/compilers) module has good collection of compilers that the tool is recognizing. If you have the ability to contribute, I would be happy to see a PR.
